MediaBrowser.Providers/Lyric/LyricManager.cs

```diff
@@ -11,60 +28,427 @@ namespace MediaBrowser.Providers.Lyric;
+
     private readonly ILyricProvider[] _lyricProviders;
     private readonly ILyricParser[] _lyricParsers;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="LyricManager"/> class.
     /// </summary>
-    /// <param name="lyricProviders">All found lyricProviders.</param>
-    /// <param name="lyricParsers">All found lyricParsers.</param>
-    public LyricManager(IEnumerable<ILyricProvider> lyricProviders, IEnumerable<ILyricParser> lyricParsers)
+    /// <param name="logger">Instance of the <see cref="ILogger{LyricManager}"/> interface.</param>
+    /// <param name="fileSystem">Instance of the <see cref="IFileSystem"/> interface.</param>
+    /// <param name="libraryMonitor">Instance of the <see cref="ILibraryMonitor"/> interface.</param>
+    /// <param name="mediaSourceManager">Instance of the <see cref="IMediaSourceManager"/> interface.</param>
+    /// <param name="lyricProviders">The list of <see cref="ILyricProvider"/>.</param>
+    /// <param name="lyricParsers">The list of <see cref="ILyricParser"/>.</param>
+    public LyricManager(
+        ILogger<LyricManager> logger,
+        IFileSystem fileSystem,
+        ILibraryMonitor libraryMonitor,
+        IMediaSourceManager mediaSourceManager,
+        IEnumerable<ILyricProvider> lyricProviders,
+        IEnumerable<ILyricParser> lyricParsers)
+    {
+        _logger = logger;
+        _fileSystem = fileSystem;
+        _libraryMonitor = libraryMonitor;
+        _mediaSourceManager = mediaSourceManager;
+        _lyricProviders = lyricProviders
+            .OrderBy(i => i is IHasOrder hasOrder ? hasOrder.Order : 0)
+            .ToArray();
+        _lyricParsers = lyricParsers
+            .OrderBy(l => l.Priority)
+            .ToArray();
+    }
+
+    /// <inheritdoc />
+    public event EventHandler<LyricDownloadFailureEventArgs>? LyricDownloadFailure;
+
+    /// <inheritdoc />
+    public Task<IReadOnlyList<RemoteLyricInfoDto>> SearchLyricsAsync(Audio audio, bool isAutomated, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(audio);
+
+        var request = new LyricSearchRequest
+        {
+            MediaPath = audio.Path,
+            SongName = audio.Name,
+            AlbumName = audio.Album,
+            ArtistNames = audio.GetAllArtists().ToList(),
+            Duration = audio.RunTimeTicks,
+            IsAutomated = isAutomated
+        };
+
+        return SearchLyricsAsync(request, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<IReadOnlyList<RemoteLyricInfoDto>> SearchLyricsAsync(LyricSearchRequest request, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        var providers = _lyricProviders
+            .Where(i => !request.DisabledLyricFetchers.Contains(i.Name, StringComparer.OrdinalIgnoreCase))
+            .OrderBy(i =>
+            {
+                var index = request.LyricFetcherOrder.ToList().IndexOf(i.Name);
```

Comment:
Maybe this `ToList` should happen outside of the `OrderBy`?

Reply:
Turns out I had already added a `.IndexOf` for `IReadOnlyList`

That's what I get from just pulling code from the SubtitleManager